core/internal/filetransfer/task_default.go

```diff
@@ -0,0 +1,93 @@
+package filetransfer
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+// DefaultTask is the default task to upload/download files
+type DefaultTask struct {
+	TaskCompletionCallback
+
+	// FileKind is the category of file being uploaded or downloaded
+	FileKind RunFileKind
+
+	// Path is the local path to the file
+	Path string
+
+	// Name is the name of the file
+	Name string
+
+	// Url is the endpoint to upload to/download from
+	Url string
+
+	// Headers to send on the upload
+	Headers []string
+
+	// Size is the number of bytes to upload
+	//
+	// If this is zero, then all bytes starting at `Offset` are uploaded; if non-zero,
+	// then that many bytes starting from `Offset` are uploaded.
+	Size int64
+
+	// Offset is the beginning of the file segment to upload
+	Offset int64
+
+	// Response is the http.Response from a successful upload or download request.
+	//
+	// This is nil for failed requests, or requests that have not completed.
+	Response *http.Response
+
+	// Error, if any.
+	Err error
+
+	// ProgressCallback is a callback to execute on progress updates
+	ProgressCallback func(int, int)
+
+	// This can be used to cancel the file upload or download if it is no longer needed.
+	Context context.Context
+}
+
+type DefaultUploadTask DefaultTask
+
+func (t *DefaultUploadTask) Execute(fts *FileTransfers) error {
+	return fts.Default.Upload(t)
+}
+func (t *DefaultUploadTask) Complete(fts FileTransferStats) {
+	t.TaskCompletionCallback.Complete(nil)
+	if fts != nil {
```

Comment:
Should it be valid to pass `nil`, or is this just for testing? If it's just for tests, could tests use a real `FileTransferStats` instead (`filetransfer.NewFileTransferStats()`)?

Reply:
yeah I was iffy about letting `TaskCompletionCallback` take in `FileTransferStats`- I think I'll just implement `Complete` on each `Task` based on the previous comment